testnet/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -0,0 +1,710 @@
+use std::io::Cursor;
+use std::time::Instant;
+
+use reqwest::blocking::{Client, RequestBuilder};
+use serde::Serialize;
+
+use secp256k1::{Secp256k1};
+
+use super::{BurnchainController, BurnchainTip};
+use super::super::operations::BurnchainOpSigner;
+use super::super::Config;
+
+use stacks::burnchains::Burnchain;
+use stacks::burnchains::bitcoin::BitcoinNetworkType;
+use stacks::burnchains::bitcoin::address::{BitcoinAddress, BitcoinAddressType};
+use stacks::burnchains::bitcoin::indexer::{BitcoinIndexer, BitcoinIndexerRuntime, BitcoinIndexerConfig};
+use stacks::burnchains::bitcoin::spv::SpvClient; 
+use stacks::burnchains::PublicKey;
+use stacks::chainstate::burn::db::burndb::BurnDB;
+use stacks::chainstate::burn::operations::{
+    LeaderBlockCommitOp,
+    LeaderKeyRegisterOp,
+    UserBurnSupportOp,
+    BlockstackOperationType,
+};
+use stacks::deps::bitcoin::blockdata::transaction::{Transaction, TxIn, TxOut, OutPoint};
+use stacks::deps::bitcoin::blockdata::opcodes;
+use stacks::deps::bitcoin::blockdata::script::{Script, Builder};
+use stacks::deps::bitcoin::network::encodable::ConsensusEncodable;
+use stacks::deps::bitcoin::network::serialize::RawEncoder;
+use stacks::deps::bitcoin::util::hash::Sha256dHash;
+use stacks::net::StacksMessageCodec;
+use stacks::util::hash::{Hash160, hex_bytes};
+use stacks::util::secp256k1::Secp256k1PublicKey;
+use stacks::util::sleep_ms;
+
+pub struct BitcoinRegtestController {
+    config: Config,
+    indexer_config: BitcoinIndexerConfig,
+    db: Option<BurnDB>,
+    chain_tip: Option<BurnchainTip>,
+}
+
+impl BitcoinRegtestController {
+
+    pub fn generic(config: Config) -> Box<dyn BurnchainController> {
+        Box::new(Self::new(config))
+    }
+
+    pub fn new(config: Config) -> Self {
+        
+        std::fs::create_dir_all(&config.node.get_burnchain_path())
+            .expect("Unable to create workdir");
+        
+        let res = SpvClient::init_block_headers(&config.burnchain.spv_headers_path, BitcoinNetworkType::Regtest);
+        if let Err(err) = res {
+            error!("Unable to init block headers: {}", err);
+            panic!()
+        }
+
+        let indexer_config = {
+            let burnchain_config = config.burnchain.clone();
+            BitcoinIndexerConfig {
+                peer_host: burnchain_config.peer_host,
+                peer_port: burnchain_config.peer_port,
+                rpc_port: burnchain_config.rpc_port,
+                rpc_ssl: burnchain_config.rpc_ssl,
+                username: burnchain_config.username,
+                password: burnchain_config.password,
+                timeout: burnchain_config.timeout,
+                spv_headers_path: burnchain_config.spv_headers_path,
+                first_block: burnchain_config.first_block,
+                magic_bytes: burnchain_config.magic_bytes
+            }
+        };
+                
```

Comment:
(Noting something that's present in `feature/neon-network-and-fixes`)

When generating BTC in regtest, Bitcoin won't allow you to spend your BTC until the block that mined them has 100 confirmations.  I got around this here by having the new controller call `build_next_block(101)` as part of its setup.

Reply:
EDIT: this should be handled by a separate driver program anyway.  Resolving...